plug.vim

```diff
@@ -968,6 +973,364 @@ while 1 " Without TCO, Vim stack is bound to explode
 endwhile
 endfunction
 
+function! s:update_python()
+python << EOF
+import datetime
+import functools
+import os
+import Queue
+import re
+import shlex
+import shutil
+import subprocess
+import tempfile
+import threading as thr
+import time
+import traceback
+import vim
+
+# Global constants, do not modify
+G_IS_WIN = vim.eval('s:is_win') == '1'
+G_MAC_GUI = vim.eval('s:mac_gui') == '1'
+G_PULL = vim.eval('s:update.pull') == '1'
+G_MAX_Y = int(vim.eval('winheight(".")'))
+G_RETRIES = int(vim.eval('get(g:, "plug_retries", 2)')) + 1
+G_TIMEOUT = int(vim.eval('get(g:, "plug_timeout", 60)'))
+G_PROGRESS = vim.eval('s:progress_opt(1)')
+
+class CmdTimedOut(Exception):
+  pass
+class CmdFailed(Exception):
+  pass
+class InvalidURI(Exception):
+  pass
+class Action(object):
+  INSTALL, UPDATE, ERROR, DONE = ['+', '*', 'x', '-']
+
+class GLog(object):
```

Comment:
@junegunn I don't mind either way. When GLog is disabled (default) I don't think it contributes to any slowdown as it returns almost immediately. I figured it would be handy later, if someone reports a py bug I can just have em pastebin the log.